src/lib/data/operations/aggregation/feature/feature-aggregation.ts

```diff
@@ -36,22 +36,26 @@ const aggregationFunctions: Record<AggregationType, Function> = {
   },
 
   [AggregationType.AVG](aggregatedFeatures: AggregatedFeatureProperties[]) {
-    return (
-      aggregationFunctions.sum(aggregatedFeatures) / aggregationFunctions.count(aggregatedFeatures)
-    );
+    let totalFeatureCount = 0;
+
+    const totalSum = aggregatedFeatures.reduce((total, feature) => {
+      totalFeatureCount += feature.featureCount;
+      return total + Number(feature.aggregatedValue) * feature.featureCount;
```

Comment:
Ok, I see, so this is now the same ponderated logic as in VL. LGTM

Reply:
Yes, it is. Using the other way, avg was really off, because of comparing "different" sets of values